src/main/scala/cromwell/engine/store/SymbolStore.scala

```diff
@@ -0,0 +1,86 @@
+package cromwell.engine.store
+
+import cromwell.binding._
+import cromwell.binding.types.WdlType
+import cromwell.binding.values.{WdlObject, WdlValue}
+import cromwell.engine.store.SymbolStore._
+
+import scala.util.{Failure, Success, Try}
+
+object SymbolStore {
+  case class SymbolStoreKey(scope: String, name: String, iteration: Option[Int], input: Boolean)
+
+  object CallInputWdlFunctions extends WdlFunctions {
+    def getFunction(name: String): WdlFunction = {
+      throw new WdlExpressionException("TODO: Some functions may be allowed in this context")
+    }
+  }
+
+  def apply(namespace: WdlNamespace, inputs: HostInputs): SymbolStore = {
+    // FIXME: Is this terminology correct?
+    val inputSymbols = inputs.map {case (name, value) => SymbolStoreEntry(name, value, input = true)}
+    val callSymbols = for {
+      workflow <- namespace.workflows
+      call <- workflow.calls
+      (k, v) <- call.inputMappings
+    } yield SymbolStoreEntry(s"${call.fullyQualifiedName}.$k", v, input = true)
+
+    new SymbolStore(inputSymbols.toSet ++ callSymbols.toSet)
+  }
+}
+
+case class SymbolStore(initialStore: Set[SymbolStoreEntry]) {
+  private var store: Set[SymbolStoreEntry] = initialStore
+
+  def locallyQualifiedInputs(call: Call): Map[String, WdlValue] = {
```

Comment:
This function really needs refactoring but it's only showing up on the diff as part of a file move. Modifying this is out of scope for this story.